bolero-engine/src/rng.rs

```diff
@@ -0,0 +1,108 @@
+use crate::{panic, Engine, SliceTestInput, TargetLocation, Test};
+use bolero_generator::driver::DriverMode;
+use core::{fmt::Debug, mem::replace};
+use rand::{rngs::StdRng, Rng, RngCore, SeedableRng};
+
+#[derive(Copy, Clone, PartialEq)]
+pub struct RngEngine {
+    pub iterations: usize,
+    pub max_len: usize,
+    pub seed: u64,
```

Comment:
Small nit; not a blocker: I wonder if `seed` should be set like proptest's [`TestRng`](https://docs.rs/proptest/0.9.4/proptest/test_runner/struct.TestRng.html)? e.g., the `thread_rng()` could be replaced with a generic parameter implementing `https://docs.rs/rand/0.7.2/rand/trait.Rng.html` with [ThreadRng](https://docs.rs/rand/0.7.2/rand/rngs/struct.ThreadRng.html) as a default.

(I'd be happy to a PR for this. It'll allow for the engine to be created once, re-used, and replayed.)

Reply:
I think we'll just stick with this for now. Feel free to open a PR!